### ISSUE TITLE
Collect judge data interactively instead of hand-editing judges.json?

### ISSUE THREAD

**Issue by anseljh** [freelawproject/judge-pics#3]
Seems we could eliminate the hand-editing-JSON step in the [contribution guidelines](https://github.com/freelawproject/judge-pics/blob/master/README.md#contributing) by adding some interactive prompts in the `convert_images.py` script, or a wrapper around it. That might make it a little easier for novices to contribute. On the other hand, the effort might better spent making a web form instead.

**Comment by johnhawkinson** [comment]
I think the pull-request workflow is sufficiently painful for images that putting any effort into a web form would be much preferable.

**Comment by mlissner** [comment]
Totally agree @johnhawkinson, and I want to move away from that process. One process I've been tinkering with is to use Github issues as a submission form. For example, this link is kind of insane, but also kind of effective:

https://github.com/freelawproject/judge-pics/issues/new?title=New%20image%20submission&body=Thank+you+for+submitting+an+image+to+the+Free+Law+Project+Judge+Pics+Collection.+This+collection+is+an+attempt+to+find+a+portrait+or+picture+of+every+United+States+judge%2C+past+and+present.%0D%0A%0D%0ABefore+you+can+submit+an+image%2C+there+is+some+housekeeping+that+we+must+take+care+of.+This+is+important+because+we+take+copyright+law+very+seriously+and+it%27s+important+that+we+get+it+right.+%0D%0A%0D%0ATo+add+an+image+to+this+collection%2C+make+sure+it+meets+our+quality+guidelines+%28https%3A%2F%2Fgithub.com%2Ffreelawproject%2Fjudge-pics%2Fblob%2Fmaster%2FREADME.md%23quality-guidelines%29%2C+attach+it+to+this+form+using+the+link+below%2C+and+complete+the+following+questions%3A%0D%0A%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0AWHO+IS+THIS%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A%0D%0A1.+Who+is+this+image+of%3F%0D%0A%0D%0A%0D%0A2.+Do+you+have+a+link+to+this+judge+in+the+CourtListener+judge+database+%28preferred%29+or+in+Wikipedia%3F%0D%0A%0D%0A%0D%0A3.+What+is+the+most+accurate+birthday+for+this+judge+%28it+can+be+accurate+to+the+day%2Fmonth%2Fyear%29%3F%0D%0A%0D%0A%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0ACREATION%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A%0D%0A1.+Who+created+this+photo+or+portrait%3F%0D%0A%0D%0A%0D%0A2.+When+was+this+item+created%3F%0D%0A%0D%0A%0D%0A3.+Where+did+this+item+come+from%3F%0D%0A%0D%0A%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0ACOPYRIGHT%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A%0D%0A1.+Is+this+your+own+work%3F+For+example%2C+did+you+take+the+photo+yourself+or+paint+the+portrait+yourself%3F%0D%0A%0D%0A%0D%0AIf+yes%2C+we+ask+that+the+photo+be+dedicated+to+the+public+domain+using+the+Creative+Commons+Zero+license+%28CC0%2C+https%3A%2F%2Fcreativecommons.org%2Fabout%2Fcc0%2F%29.+To+dedicate+your+image%28s%29+to+the+public+domain+under+the+CC0+license%2C+state+%22I+AGREE%22%3A%0D%0A%0D%0A%0D%0A%0D%0A2.+If+this+is+not+your+own+work%2C+is+it+already+in+the+public+domain%3F+Check+one+of+the+boxes+below+if+so%3A%0D%0A%0D%0A+-+%5B+%5D+Work+of+a+United+Stated+federal+agency%0D%0A+-+%5B+%5D+First+published+in+the+United+States+before+1923%0D%0A+-+%5B+%5D+Photo+of+a+two-dimensional+work+whose+author+died+more+than+100+years+ago%0D%0A%0D%0A%0D%0A3.+If+you+did+not+create+this+image+yourself+and+it+is+not+in+the+public+domain%2C+is+there+another+reason+you+think+this+item+is+free+from+copyright%3F%0D%0A%0D%0AThank+you+for+contributing%21+We+hope+this+has+been+relatively+painless%2C+and+we%27ll+get+to+your+submission+soon

Oddly, bit.ly and goo.gl both think it's an invalid link, but it does seem to work, providing a form for you to fill out and a way you can drag and drop your photos into Github. One issue is that it crashes github if you're not logged in. Another is that it's an insane link.

Two other thoughts:

1. I'd be happy to have this built into CourtListener, but doing so is complicated...we've never allowed people to submit photos, so somebody would have to figure that out.

2. Assuming that's too hard to be worth it, another idea is to find some other website that allows easy submissions like this, and point people to that.

Where this needs to land is that anybody with two wits can do a submission without knowing the first thing about code.

**Comment by mlissner** [comment]
The other reason we haven't moved forward on this process in a material way is that we need to be very careful about copyright. One stupid infraction and we're sunk. Ideally, we find a way for DMCA to protect us, but if we're reviewing people's submissions I don't think we can even use that shield.

cc: @brianwc

**Comment by anseljh** [comment]
@mlissner I think the feature you're looking for is issue templates: https://github.com/blog/2111-issue-and-pull-request-templates

**Comment by mlissner** [comment]
@anseljh Yeah, but those always apply to every issue in a repo, which isn't what we want.

**Comment by johnlettman** [comment]
> [@anseljh](https://github.com/anseljh) Yeah, but those always apply to every issue in a repo, which isn't what we want.

Not exactly.

You can define multiple for the same project. 
[Here is an example in one of my repositories.](https://github.com/johnlettman/imagenex831l/tree/main/.github/ISSUE_TEMPLATE)

<img width="831" height="271" alt="Image" src="https://github.com/user-attachments/assets/7282acd3-b969-405f-8e87-327818ed59f8" />

---

@mlissner I have created a variation of yours that could be inserted into `.github/ISSUE_TEMPLATE`. GitHub will retain the "blank" issue, or you could alternatively snowball this example into "bug" or "feature request" issues.

[judge_submission.md](https://github.com/user-attachments/files/25277925/judge_submission.md)
